apis/python/notebooks/tutorial_soma_shape.ipynb

```diff
@@ -0,0 +1,1676 @@
+    }
+   ],
+   "source": [
+    "exp.ms[\"RNA\"].X[\"data\"].maxshape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f288b733-8415-4b73-8237-08f5704f0586",
+   "metadata": {},
+   "source": [
+    "The other N-dimensional arrays are similar:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "96523a2b-10d3-4e2c-b76e-1b6e814c8774",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['X_draw_graph_fr', 'X_pca', 'X_tsne', 'X_umap']"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "obsm = exp.ms[\"RNA\"].obsm\n",
+    "list(obsm.keys())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "82b16ded-298c-4d7e-8dfd-ffb4b36c37c6",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['connectivities', 'distances']"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "obsp = exp.ms[\"RNA\"].obsp\n",
+    "list(obsp.keys())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "bbe56e08-c237-48df-8b0d-393057e7e6fa",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[(2638, 50), (9223372036854773759, 9223372036854773759)]"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "[\n",
+    "    obsm[\"X_pca\"].shape,\n",
+    "    obsm[\"X_pca\"].maxshape,\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "7577221c-85c7-4549-847d-8cbcc1b771ab",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[(2638, 2638), (9223372036854773759, 9223372036854773759)]"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "[\n",
+    "    obsp[\"distances\"].shape,\n",
+    "    obsp[\"distances\"].maxshape,\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f7c0e4bb-30fd-4b24-9231-e72c79d0a1c2",
+   "metadata": {},
+   "source": [
+    "In particular, the `X` array in this experiment -- and in most experiments -- is _sparse_. That means there needn't be a number in every row or cell of the matrix. Nonetheless, the shape serves as a soft limit for reads and writes: you'll get an exception trying to read or write outside of these."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aecfff79-d5ac-4361-ba56-0c5cad05206d",
+   "metadata": {},
+   "source": [
+    "As a convenience, you can see all the experiment's objects' shapes at once as follows:\n",
+    "\n",
+    "```\n",
+    "import tiledbsoma.io\n",
+    "tiledbsoma.io.show_experiment_shapes(exp.uri)\n",
+    "```"
```

Comment:
Should this code reside within a code cell?

Reply:
Maybe I can add a smaller dataset (like `pbmc3k_unprocessed` which is just `obs` `var` `X` and not much else) ...

Reply:
OK np - fine as is then.